### PR TITLE
Migrate bucket upload workflow to GitLab

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Description
- Removes upload dev rules that would have jobs not run when `MARKET_GCS_BUCKET` equaled `marketplace-dist-dev`
- Removes assignment of `MARKET_GCS_BUCKET` environment variable to `marketplace-dist-dev` in various job configurations
- Changes the default value of `MARKET_GCS_BUCKET` to the production value (AKA `marketplace-dist`)
- Removes the scheduled bucket-upload workflow from the CircleCI configuration

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [x] 6.0.0
- [x] 6.1.0
- [x] 6.2.0

## Does it break backward compatibility?
   - [x] Maybe

## Must have
- [x] ~~Tests~~ N/A
- [x] ~~Documentation~~ N/A

### CODE REVIEW

**Comment by avidan-H** [comment]
I'm adding the `docs-approved` label since there isn't any documentation for review.

**Comment by avidan-H** [comment]
@DinaMeylakh @guyfreund I would appreciate it if you could take a look at this PR seeing as the two of you are far more familiar with this flow than myself - I easily could be missing/forgetting something here.

**Review by guyfreund — COMMENTED** [review]
Looks good to me. Do we plan to remove all additional infra on `config.yml` once we see that `GitLab` is stable enough?

**Comment by Itay4** [comment]
> Looks good to me. Do we plan to remove all additional infra on `config.yml` once we see that `GitLab` is stable enough?

correct

File: .circleci/config.yml [hunk 1]
```diff
@@ -1610,18 +1610,6 @@ workflows:
...
-          # should trigger every day at 9 AM & 9 PM UTC (11:00 PM & 11:00 AM Israel Time)
-          cron: "0 9,21 * * *"
-          filters:
-            branches:
-              only:
-                - master
-    jobs:
-      *bucket_upload_jobs
-
   bucket_upload_trigger:
```
Comment (DinaMeylakh) [inline .circleci/config.yml:1613]: Do we want to keep this?

**Review by DinaMeylakh — CHANGES_REQUESTED** [review]
allow Utils/gitlab_triggers/trigger_upload_flow_build.sh to use marketplace-dist also

File: .circleci/config.yml (hunk 1 above)
Comment (avidan-H) [reply .circleci/config.yml:1613]: I think this falls under the category of all the remaining infrastructure we would need to still remove pending seeing that the flow operates in `GitLab` stably as @guyfreund mentioned in his comment https://github.com/demisto/content/pull/14130#pullrequestreview-727090083.

**Review by DinaMeylakh — APPROVED** [review]

**Review by guyfreund — APPROVED** [review]

**Comment by guyfreund** [comment]
good luck to us all

**Review by Noy-Maimon — APPROVED** [review]

**Comment by Itay4** [comment]
@Noy-Maimon @guyfreund merging this - fyi

**Pull request merged**